es/18_4_Accediendo_a_Bitcoind_con_Python.md

```diff
@@ -0,0 +1,463 @@
+```
+## Crear una dirección
+
+Crear una nueva dirección con Python 3 solo requiere el uso de un RPC como `getnewaddress` o `getrawchangeaddress`.
+```
+new_address = rpc_client.getnewaddress("Learning-Bitcoin-from-the-Command-Line")
+new_change_address = rpc_client.getrawchangeaddress()
+```
+En este ejemplo, se da el comando `getnewaddress` un argumento: la etiqueta `Learning-Bitcoin-from-the-Command-Line` .
+
+## Enviar una transacción
+
+Crear una transacción en Python 3 requiere combinar algunos de los ejemplos anteriores (de crear direcciones y recuperar UTXOs) con algunos nuevos comandos RPC para crear, firmar y enviar una transacción mucho de lo que ha hecho anteriormente desde la línea de comandos.
+Hay cinco pasos a seguir:
+
+0. Crear dos direcciones, una que actuará como receptor y la otra para el cambio.
+1. Seleccione un UTXO y establezca los detalles de la transacción.
+2. Crear una transacción en bruto.
+3. Firmar la transacción en bruto con la clave privada de la UTXO.
+4. Difundir la transacción en el bitcoin testnet.
+
+### 1. Seleccione UTXO y definir los detalles de la transacción
+
+En el siguiente fragmento de código, primero seleccione el UTXO que queremos gastar. Entonces usted consigue su dirección, la identificación de la transacción, y el índice vectorial de la salida.
+```py
+utxos = rpc_client.listunspent()
+selected_utxo = utxos[0]  # again, selecting the first utxo here
+utxo_address = selected_utxo['address']
+utxo_txid = selected_utxo['txid']
+utxo_vout = selected_utxo['vout']
+utxo_amt = float(selected_utxo['amount'])
+```
+A continuación, también recupera la dirección del destinatario a la que quiere enviar los bitcoins, calcula la cantidad de bitcoins que quiere enviar, y calcula la tarifa minera y la cantidad de cambio. Aquí, la cantidad se divide arbitrariamente en dos y una cuota de minero se establece arbitrariamente.
+```py
+recipient_address = new_address
+recipient_amt = utxo_amt / 2  # sending half coins to recipient
+miner_fee = 0.00000300        # choose appropriate fee based on your tx size
+change_address = new_change_address
+change_amt = float('%.8f'%((utxo_amt - recipient_amt) - miner_fee))
+```
+> :warning: **WARNING:** Obviamente un programa real tomaría decisiones más sofisticadas sobre qué UTXO usar, qué hacer con los fondos, y qué cuota de minero pagar.
+
+### 2. Crear transacciones sin procesar
+
+Ahora usted tiene toda la información para enviar una transacción, pero antes de que pueda enviar una, tiene que crear una transacción.
+```py
+txids_vouts = [{"txid": utxo_txid, "vout": utxo_vout}]
+addresses_amts = {f"{recipient_address}": recipient_amt, f"{change_address}": change_amt}
+unsigned_tx_hex = rpc_client.createrawtransaction(txids_vouts, addresses_amts)
+```
+Recuerde que el formato del comando `createrawtransaction` es:
+`$ bitcoin-cli createrawtransaction '[{"txid": <utxo_txid>, "vout": <vector_id>}]' '{"<address>": <amount>}'`
```

Comment:
Line 371 shows up on same line as 370, add another empty line between 370 and 371 for 371 to show up on it's own line